test/testdata/test.sh

```diff
@@ -23,6 +23,7 @@ function test_project {
 
   header_text "Performing tests in dir $1"
   pushd "$(dirname "$0")/../../testdata/$1"
+  rm -rf test/
```

Comment:
This check ensures that the testdata samples are updated with the latest changes.
Why are we removing tests/ dir from here?

Reply:
if the test dir is not removed 
then scaffolded e2e tests will run for all the sample which we don't require right? we only want to test one sapmle(project-v4)

Reply:
The testdata check if all samples are == they should be with the latest changes.
We are changing the default scaffold so all v4 samples should have the tests now.

We do not need to execute and run the e2e tests against all projects
If we see that does not take too long that is fine would be great. But if we see that take too long then we just test 1/2 samples. But that is related to the new GitHub action and not the change test data check.

is that make sense?

Reply:
Why the e2e tests will run for all samples?
We are calling make test, right?

Also, see that it is not a problem to run the e2e tests for all samples (unless it would take too long)
The idea to run for one for then is to simplify the implementation.